tests/installation/welcome.pm

```diff
@@ -42,80 +73,66 @@ sub run {
     ensure_fullscreen;
 
     # Process expected pop-up windows and exit when welcome/beta_war is shown or too many iterations
-    while ($iterations++ < scalar(@welcome_tags)) {
-        # See poo#19832, sometimes manage to match same tag twice and test fails due to broken sequence
-        wait_still_screen 5;
-        assert_screen(\@welcome_tags, 500);
-        # Normal exit condition
-        if (match_has_tag 'inst-betawarning' || match_has_tag 'inst-welcome' || match_has_tag 'inst-welcome-no-product-list') {
-            last;
-        }
-        if (match_has_tag 'scc-invalid-url') {
-            die 'SCC reg URL is invalid' if !get_var('SCC_URL_VALID');
-            send_key 'alt-r';    # registration URL field
-            send_key_until_needlematch 'scc-invalid-url-deleted', 'backspace';
-            type_string get_var('SCC_URL_VALID');
-            wait_still_screen 2;
-            # Press Ok to confirm scc url
-            wait_screen_change { send_key 'alt-o' };
-            next;
-        }
-        if (match_has_tag 'inst-welcome-confirm-self-update-server') {
-            wait_screen_change { send_key $cmd{ok} };
-            next;
-        }
-        if (match_has_tag 'no-product-found-on-scc') {
-            record_soft_failure 'bsc#1056413';
-            wait_screen_change { send_key $cmd{ok} };
-            next;
-        }
-        if (match_has_tag('untrusted-ca-cert')) {
-            send_key 'alt-t';
-            wait_still_screen 5;
-            next;
-        }
-        if (match_has_tag 'linuxrc-dhcp-question') {
-            send_key 'ret';
-        }
-    }
+    handle_popups while ($iterations++ < scalar(@welcome_tags));
+}
 
-    # Process beta warning if expected
-    if ($expect_beta_warn) {
-        assert_screen 'inst-betawarning';
-        wait_screen_change { send_key 'ret' };
-    }
+sub process_beta_warning_if_expected {
```

Comment:
we have function which suppose to handle all popups, why we need a separate function to handle beta popup ? why not to do it all in one place ?

Reply:
to make it clear from product perspective that we do something different here